### PR TITLE
Audio: add structure for siren detection

### PR DESCRIPTION
**Pull request by HongyiSun**

### CODE REVIEW

**Comment by Capri2014** [comment]
@HongyiSun try to rebase and avoid commits info like "Merge branch 'master' into master"

**Comment by HongyiSun** [comment]
> @HongyiSun try to rebase and avoid commits info like "Merge branch 'master' into master"

Feel free to squash it into one commit

File: modules/audio/inference/siren_detection.cc [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+  for (const auto& channel : signals) {
+    for (const auto& i : channel) {
+      *data++ = i;
+    }
+  }
+
+  torch::Tensor torch_input = torch::from_blob(data, {6, 1, 30075});
+  std::vector<torch::jit::IValue> torch_inputs;
+  torch_inputs.push_back(std::move(torch_input.to(device_)));
```
Comment (storypku) [inline modules/audio/inference/siren_detection.cc:57]: AIUI, the temporary value returned by `torch_input.to(device_)` is  `rvalue` reference. No need to call `std::move` explicitly here.

File: modules/audio/common/audio_gflags.cc [hunk 2]
```diff
@@ -17,3 +17,6 @@
 #include "modules/audio/common/audio_gflags.h"
 
 DEFINE_int32(cache_signal_time, 3, "The time to cache signal");
+DEFINE_string(torch_siren_detection_model,
+              "/apollo/modules/audio/data/siren_detection_model.pt",
+              "Siren detection model file");
```
Comment (xiaoxq) [inline modules/audio/common/audio_gflags.cc:22]: Use constexpr variable if it's only used in one place and almost never change.
https://abseil.io/tips/45

File: modules/audio/inference/siren_detection.cc [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include "modules/audio/inference/siren_detection.h"
+
+#include <vector>
+#include <utility>
+
+#include <omp.h>
```
Comment (xiaoxq) [inline modules/audio/inference/siren_detection.cc:22]: Put C headers before C++ headers.

File: modules/audio/common/audio_gflags.cc (hunk 2 above)
Comment (HongyiSun) [reply modules/audio/common/audio_gflags.cc]: will change those flags for model path together...

**Pull request merged**